tool/tsh/common/kubectl.go

```diff
@@ -452,12 +452,18 @@ func makeAndStartKubeLocalProxy(cf *CLIConf, config *clientcmdapi.Config, cluste
 	return closeFn, localProxy.KubeConfigPath(), nil
 }
 
+// profileRequireKubeLocalProxy returns true if provided profile indicates a
+// local proxy is required for kube access.
+func profileRequireKubeLocalProxy(profile *profile.Profile) bool {
+	return profile.KubeProxyAddr == profile.WebProxyAddr && profile.TLSRoutingConnUpgradeRequired
+}
```

Comment:
Can we refactor this into profile method?

`profile.RequiresKubeLocalProxy()` ?